controllers/storage.go

```diff
@@ -432,6 +436,10 @@ func (s *Storage) Archive(artifact *sourcev1.Artifact, dir string, filter Archiv
 		return err
 	}
 
+	if s.ArtifactMaxSize > 0 && sz.written > s.ArtifactMaxSize {
+		return fmt.Errorf("artifact size %d exceeds the max limit of %d bytes", sz.written, s.ArtifactMaxSize)
+	}
+
```

Comment:
Is the writerCounter incremented mid-flight as a stream? If so, we could short-circuit and stop the operation before it goes beyond the pre-set limit within `Walk`.

Reply:
Yes but my intention here was to tell users which is the final size of the artifact.

Reply:
I would short circuit the process instead, as otherwise half of the usefulness of the bound is lost.

Reply:
If I move the check inside `Walk` then `sz.written` is always zero.